### PR TITLE
Lab3

### PR DESCRIPTION
**Pull request by VadimDomantsevich**

### CODE REVIEW

File: Lab2/BLL/Interfaces/IService.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        Task Create(T item);
+
+        Task Delete(int id);
+
+        Task Update(T item);
+
+        Task<T> GetById(int id);
+
+        Task<IEnumerable<T>> GetAll();
```
Comment (alexey62soft) [inline Lab2/BLL/Interfaces/IService.cs]: Return List or Array instead of IEnumerable.
FIx in all methods.

File: Lab2/BLL/Models/Student.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        public int Id { get; set; }
+
+        public string FullName { get; set; }
+
+        public string RecordNumber { get; set; }
+
+        public int GroupId { get; set; }
+
+        public string PhoneNumber { get; set; }
+
```
Comment (alexey62soft) [inline Lab2/BLL/Models/Student.cs]: Useless line

File: Lab2/DAL/EntityFramework/GeneralRepository.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            await _dBContext.SaveChangesAsync();
+        }
+
+        public async Task Delete(T item)
+        {
+            _dBContext.Set<T>().Remove(item);
+            await _dBContext.SaveChangesAsync();
+        }
+
+        public async Task<IEnumerable<T>> GetAll()
```
Comment (alexey62soft) [inline Lab2/DAL/EntityFramework/GeneralRepository.cs]: Not return IEnumerable

File: Lab2/UI/Services/GroupsBaseConsoleService.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
+﻿using BLL.Infrastructure;
+using BLL.Models;
+using System;
+using System.Threading.Tasks;
+using UI.Interfaces;
+using UI.Services.PrintConsoleService;
+
+namespace UI.Services.MainConsoleService
```
Comment (alexey62soft) [inline Lab2/UI/Services/GroupsBaseConsoleService.cs]: Fix namespace

File: Lab2/UI/Services/GroupsCrudConsoleService.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
+﻿using BLL.Models;
+using BLL.Services;
+using System;
+using System.Threading.Tasks;
+using UI.Helpers;
+using UI.Interfaces;
+
+namespace UI.Services.CrudConsoleService
```
Comment (alexey62soft) [inline Lab2/UI/Services/GroupsCrudConsoleService.cs]: Fix namespaces in all console services

**Review by alexey62soft — CHANGES_REQUESTED** [review]

File: Lab2/UI/Services/GroupsCrudConsoleService.cs (hunk 5 above)
Comment (VadimDomantsevich) [reply Lab2/UI/Services/GroupsCrudConsoleService.cs]: Fixed

File: Lab2/UI/Services/GroupsBaseConsoleService.cs (hunk 4 above)
Comment (VadimDomantsevich) [reply Lab2/UI/Services/GroupsBaseConsoleService.cs]: Fixed

File: Lab2/DAL/EntityFramework/GeneralRepository.cs (hunk 3 above)
Comment (VadimDomantsevich) [reply Lab2/DAL/EntityFramework/GeneralRepository.cs]: Fied

File: Lab2/BLL/Models/Student.cs (hunk 2 above)
Comment (VadimDomantsevich) [reply Lab2/BLL/Models/Student.cs]: Fixed

File: Lab2/BLL/Interfaces/IService.cs (hunk 1 above)
Comment (VadimDomantsevich) [reply Lab2/BLL/Interfaces/IService.cs]: Fixed

**Review by alexey62soft — APPROVED** [review]
Report also should be there.

**Pull request merged**